src/main/groovy/life/qbic/portal/sampletracking/web/views/samplefile/UploadComponent.groovy

```diff
@@ -76,11 +76,15 @@ class UploadComponent extends VerticalLayout {
         return new SucceededListener() {
             @Override
             void uploadSucceeded(SucceededEvent event) {
-
                 Set<String> sampleIds = [] as Set
-
+                String separator = ","
                 try {
-                    sampleIds = new String(uploadContent.toByteArray()).split("\n") as Set
+                    Set inputLines = new String(uploadContent.toByteArray()).split("\n") as Set
+                    for (line in inputLines) {
+                        if (line.capitalize().startsWith("Q")) {
+                            sampleIds.add(line.split(separator)[0])
+                        }
```

Comment:
I am not sure that we want to change the input by capitalizing and filtering for `Q*`
If there is an error in the file it should fail or notify the user instead.
```suggestion
                        if (line.startsWith("Q")) {
                            sampleIds.add(line.split(separator)[0])
                        } else {
                            //TODO inform the customer or log or sth :D 
                        }
```